lib/src/tree_cursor.c

```diff
@@ -291,3 +291,12 @@ const char *ts_tree_cursor_current_field_name(const TSTreeCursor *_self) {
     return NULL;
   }
 }
+
+TSTreeCursor ts_tree_cursor_copy(const TSTreeCursor *_cursor) {
+    const TreeCursor *cursor = (const TreeCursor *)_cursor;
+    TSTreeCursor res = {NULL, NULL, {0, 0}};
+    TreeCursor *copy = (TreeCursor *)&res;
+    copy->tree = cursor->tree;
+    array_push_all(&copy->stack, &cursor->stack);
+    return res;
```

Comment:
It looks like this has a different indentation than the other functions. Could you adjust it to match?

Reply:
BTW shouldn't we increment ref-count of the tree here?

Reply:
The cursor doesn't seem to retain it at all. Probably that's fine